### ISSUE TITLE
Ninjacart (BA: AGR10) - compliance check

### ISSUE THREAD

**Issue by sahil-ondc** [ONDC-Official/v1.2.0-logs#1882]
### on_search
The Buyer App should send a 'NACK' if the following errors are being received from the 'Seller-App':
 
- "/message/catalog/bpp~1descriptor/tags/0 must NOT have additional properties (display)"
- "/message/catalog/bpp~1providers/0/items/0 must have required property 'category_id'"
- "/message/catalog/bpp~1providers/0/items/0/quantity/unitized/measure/value should be a number in string format"
- "/message/catalog/bpp~1providers/0/items/0/quantity/available/count available count should be a number in string format"
- "/message/catalog/bpp~1providers/0/items/0/quantity/maximum/count maximum count should be a number in string format"
 
### on_init
- There should be a 'NACK' for an invalid tax number provided in the /message/order/tags/0/list/0/value.
 
### on_status_pending
- There should be a 'NACK' if the RoutingTag object is not present in the tags of the 'Delivery' object.
- There should be a 'NACK' if the available count is less than @ondc/org/item_quantity.count.
 
### on_select_out_of_stock
The Buyer App should send a 'NACK' for the incorrect 'DOMAIN-ERROR' message received from the 'Seller-App'.
 
## Flow 4, 5 & 6
Same types of errors are processed in other flows as well.

@anwar-ninja

**Comment by anwar-ninja** [comment]
@sahil-ondc **.There should be a 'NACK' if the available count is less than @ondc/org/item_quantity.count. can u please elaborate this point.

**Comment by anwar-ninja** [comment]
@sahil-ondc

**Comment by anwar-ninja** [comment]
https://github.com/ONDC-Official/v1.2.0-logs/pull/1918 resolved please check

**Comment by sahil-ondc** [comment]
## Flow 1

### search (inc catalog)

- Is one time incremental search supported ?

## Other flows

### confirm

- accept_bpp_terms should not be provided as those are not enabled yet

### on_confirm

- /message/order/fulfillments/0/end must have required property 'person' (BNP should NACK)

### on_status

- in on_confirm (state is already Accepted & Pending) why on_status has been provided and updated_at has been updated without any change (should be handled at BNP end)

@anwar-ninja

**Comment by anwar-ninja** [comment]
yes incremental supported. updated pr: https://github.com/ONDC-Official/v1.2.0-logs/pull/1942

**Comment by anwar-ninja** [comment]
https://github.com/ONDC-Official/v1.2.0-logs/pull/1979 fixed
